service/src/test/java/bio/terra/profile/service/profile/flight/update/UpdateProfileFlightTest.java

```diff
@@ -0,0 +1,231 @@
+
+    when(samService.hasActions(userRequest, SamResourceType.PROFILE, profile.id()))
+        .thenReturn(true);
+    when(profileDao.getBillingProfileById(profile.id()))
+        .thenReturn(profile); // first call occurs pre-flight
+    when(profileDao.getBillingProfileById(profile.id()))
+        .thenReturn(updatedProfile); // second call occurs post-update
+    when(profileDao.updateProfile(profile.id(), null, newBillingAccount)).thenReturn(true);
+
+    var result =
+        profileService.updateProfile(
+            profile.id(),
+            new UpdateProfileRequest().billingAccountId(newBillingAccount),
+            userRequest);
+    assertEquals(profile.id(), result.getId());
+    assertEquals(newBillingAccount, result.getBillingAccountId());
+
+    verify(samService)
+        .verifyAuthorization(
+            userRequest, SamResourceType.PROFILE, profile.id(), SamAction.UPDATE_BILLING_ACCOUNT);
+    verify(samService, times(0))
+        .verifyAuthorization(
+            userRequest, SamResourceType.PROFILE, profile.id(), SamAction.UPDATE_METADATA);
+    verify(billingCow).testIamPermissions(any());
+  }
+
+  @Test
+  void updateProfileDescriptionSuccess() throws InterruptedException {
+    String newDescription = "newDescription";
+
+    var profile = ProfileFixtures.createGcpBillingProfile("ABCD1234");
+    var updatedProfile =
+        new BillingProfile(
+            profile.id(),
+            profile.displayName(),
+            newDescription,
+            profile.biller(),
+            profile.cloudPlatform(),
+            profile.billingAccountId(),
+            profile.tenantId(),
+            profile.subscriptionId(),
+            profile.managedResourceGroupId(),
+            profile.createdTime(),
+            profile.lastModified(),
+            profile.createdBy());
+
+    when(samService.hasActions(userRequest, SamResourceType.PROFILE, profile.id()))
+        .thenReturn(true);
+    when(profileDao.getBillingProfileById(profile.id()))
+        .thenReturn(profile); // first call occurs pre-flight
+    when(profileDao.getBillingProfileById(profile.id()))
+        .thenReturn(updatedProfile); // second call occurs post-update
+    when(profileDao.updateProfile(profile.id(), newDescription, null)).thenReturn(true);
+
+    var result =
+        profileService.updateProfile(
+            profile.id(), new UpdateProfileRequest().description(newDescription), userRequest);
+    assertEquals(profile.id(), result.getId());
+    assertEquals(newDescription, result.getDescription());
+
+    verify(samService)
+        .verifyAuthorization(
+            userRequest, SamResourceType.PROFILE, profile.id(), SamAction.UPDATE_METADATA);
+    verify(samService, times(0))
+        .verifyAuthorization(
+            userRequest, SamResourceType.PROFILE, profile.id(), SamAction.UPDATE_BILLING_ACCOUNT);
+    verifyNoInteractions(crlService);
+  }
+
+  @Test
+  void updateProfileUnauthorizedFailure() throws InterruptedException {
```

Comment:
I also opted not to test every possible authz check failure, but could be convinced to add them in